packages/mongodb-rag-core/src/eval/getConversationEvalCasesFromYaml.ts

```diff
@@ -1,19 +1,52 @@
 import yaml from "yaml";
 import { z } from "zod";
 
-export const ConversationEvalCaseSchema = z.object({
-  name: z.string(),
-  expectation: z
+/**
+  Zod schema loosely based on the OpenAI.FunctionDefinition interface.
+  Validates input tool definitions passed to generateResponse.
+ */
+export const ToolDefinitionSchema = z.object({
+  name: z.string().describe("The name of the function to be called."),
+  description: z
     .string()
     .optional()
+    .describe("A description of what the function does."),
+  parameters: z
+    .record(
+      z.string(),
+      z.object({
+        type: z.string().optional(),
+        description: z.string().optional(),
+      })
```

Comment:
i dont think this is correct tyling b/c it can be deeply nested, have additional props like min, max, etc. 

instead i think just z.any() is fine here

Reply:
unless you can find a zod type that represents all valid json schema (which might exist given popularity of both tools)....don't sweat this tho. z.any() should be fine